### ISSUE TITLE
Empty Count Violation should not warn if variable is Int

### ISSUE THREAD

**Issue by dlpgithub** [realm/SwiftLint#3433]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Not sure if iam right but
Empty Count Violation should not warn if variable is Int

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
Empty Count Violation: Prefer checking `isEmpty` over comparing `count` to zero. (empty_count)

i would prefer checking... but:

Value of type 'Int' has no member 'isEmpty'
```

### Environment

* SwiftLint version: 0.41.0
* Installation method used: CocoaPods

* Which Xcode version are you using: 12.2

**Comment by asolovev** [comment]
I think the main reference to this problem is #827 . There is some discussion there and this applies to multiple item types that lead to false positives.
From what I understand the only workaround right now is to disable this rule manually in some cases. If count is often used, an extension can be created to add `isEmpty` variable which will require disabling the rule in the extension only.

**Comment by lordzsolt** [comment]
I presume you are getting this warning, because you have a variable named `count`.

There's a [configuration for it](https://realm.github.io/SwiftLint/empty_count.html).
You can set `only_after_dot: true`, so only `.count` will raise a violation.

This should solve most problems, albeit not the ones mentioned by @asolovev

**Comment by cbiggin** [comment]
This just happened to me and yes:
- I have a variable (in a struct) called `count`
- setting `only_after_dot: true` fixed it

It's still a bug though and I don't agree that adding an extension of `isEmpty` is acceptable.

**Comment by shantpanos** [comment]
Setting `only_after_dot: true` won't fix the cases where `.count` is still of `Int` type. For example:
```swift
while tracks.count =! 0 { ... }
```

**Comment by SimplyDanny** [comment]
It's a false positive, but one that we have to accept since SwiftLint doesn't have type information at hand. Instead, it works on the syntax level only. So counter examples like mentioned in this issue can appear and need to be ignored using `// swiftlint:disable` commands.
